### ISSUE TITLE
Add a method for length∘skipmissing

### ISSUE THREAD

**Issue by bkamins** [JuliaLang/julia#42672]
This currently fails:
```
julia> x = [1, 2, 3]
3-element Vector{Int64}:
 1
 2
 3

julia> (length∘skipmissing)(x)
ERROR: MethodError: no method matching length(::Base.SkipMissing{Vector{Int64}})
```
I propose to add an appropriate method even if this is an O(n) operation since it is quite commonly needed.

CC @nalimilan

**Comment by timholy** [comment]
Can't this be `count(!ismissing, x)`

**Comment by timholy** [comment]
Can't this be `count(!ismissing, x)`

**Comment by bkamins** [comment]
No, because only `x -> count(!ismissing, x)` is something you can dispatch on and it is an anonymous function that is dynamically typed as opposed to `length∘skipmissing` which has a known type.
If we had an option to curry `count(!ismissing)` then indeed `length∘skipmissing` would not be needed.

**Comment by bkamins** [comment]
This is the idea. I want to add `Base.length(x::Base.SkipMissing)` definition although it is O(n).

**Comment by bkamins** [comment]
Ah - good point since:
```
julia> typeof(length∘skipmissing)
ComposedFunction{typeof(length), typeof(skipmissing)}
```
is callable so the title was wrong.

**Comment by bkamins** [comment]
Actually sometimes one gets `y = skipmissing(x)` as input, and then there is a question how to count number of elements of `y`, would `count(v -> true, y)` be the recommended pattern?
